estate/models/estate_property.py

```diff
@@ -0,0 +1,110 @@
+from odoo import api, fields, models, tools
+from odoo.exceptions import UserError, ValidationError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Property"
+    _order = "id desc"
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)',
+         'A property expected price must be strictly positive'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)',
+         'The selling price must be positive'),
+    ]
+    
+    name = fields.Char("Title", required=True)
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("canceled", "Canceled"),
+        ],
+        default="new",
+        required=True,
+        copy=False,
+        string="Status"
+    )
+
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(
+        "Available From",
+        copy=False, default=fields.Datetime.add(fields.Datetime.today(), months=3)
+    )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ]
+    )
+    
+    property_type_id = fields.Many2one("estate.property.type",string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    salesman_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    
+    total_area = fields.Float("Total Area (sqm)", compute="_compute_area")
+    best_price = fields.Float("Best Offer", compute="_compute_best_price")
+    
+    @api.depends("living_area", "garden_area")
+    def _compute_area(self) -> None:
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+             
+    @api.depends("offer_ids")
+    def _compute_best_price(self) -> None:
+        for property in self:
+            property.best_price = max(property.offer_ids.mapped('price')) if property.offer_ids else 0
+            
+    @api.constrains("selling_price")
+    def _check_selling_price(self) -> None:
+        # Filter based on selling price because when we cancel a approved offer, the price is reset to 0
```

Comment:
```suggestion
 
```
If you feel the need to comment, in most cases, it means you should write code that is more clear